kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -362,4 +377,42 @@ public void deleteAllowed(Subscriber subscriber) {
             subscriptionService.deletePreconditionCheck(subscription);
         }
     }
+
+    @Override
+    public void deactivateAllSubscriptionsForSubscriber(long callingNumber) {
+        LOGGER.info("Recieved Release Number {} for Deactivation.", callingNumber);
+        Subscriber subscriberByMsisdn = this.getSubscriber(callingNumber);
+        if (subscriberByMsisdn == null) {
+            LOGGER.info("Subscriber for msisdn {} is not found." , callingNumber);
+            throw new IllegalArgumentException(String.format(KilkariConstants.SUBSCRIBER_NOT_FOUND, callingNumber));
+        }
+        LOGGER.info("Found Subscriber for msisdn {} .", callingNumber);
+        int counter = 0;
+        for (Subscription subscription : subscriberByMsisdn.getAllSubscriptions()) {
+            if ((subscription.getStatus() == SubscriptionStatus.PENDING_ACTIVATION) || (subscription.getStatus() == SubscriptionStatus.ACTIVE) || (subscription.getStatus() == SubscriptionStatus.HOLD)) {
+                try {
+                    LOGGER.info("Deactivating Subscription with Id {} for msisdn.", subscription.getSubscriptionId());
+                    subscriptionService.deactivateSubscription(subscription, DeactivationReason.WEEKLY_CALLS_NOT_ANSWERED);
+                    deactivationAuditService.auditSuccess(subscription.getSubscriptionId(), subscriberByMsisdn.getId(), subscription.getOrigin(), subscription.getStatus(), callingNumber);
+                    counter++;
+                } catch (Exception e) {
+                    LOGGER.error(String.format("Unexpected exception in deactivating subscription %s: %s", subscription.getSubscriptionId(), ExceptionUtils.getFullStackTrace(e)));
+                    deactivationAuditService.auditFailure(subscription.getSubscriptionId(), subscriberByMsisdn.getId(), subscription.getOrigin(), subscription.getStatus(), callingNumber, ExceptionUtils.getFullStackTrace(e));
+                    throw new IllegalStateException(e);
+                }
+            }
+        }
+        LOGGER.info("Deactivated {} Subscritions for msisdn {}.", counter, callingNumber);
+        // Add callingNumber to WeeklyCallsNotAnsweredMsisdnRecord (tableName = "nms_blocked_msisdn")
+        BlockedMsisdnRecord record = blockedMsisdnRecordDataService.findByNumber(callingNumber);
+        //TODO: we can use createOrUpdate method of MotechDataService once the bug is fixed.
+        if (record == null) {
+            blockedMsisdnRecordDataService.create(new BlockedMsisdnRecord(callingNumber, DeactivationReason.WEEKLY_CALLS_NOT_ANSWERED));
+        } else {
+            record.setDeactivationReason(DeactivationReason.WEEKLY_CALLS_NOT_ANSWERED);
```

Comment:
Suggestion: you might want to have a debug log here that says you found an existing existing and are updating it or change the logger line below to say added or updated depending on the operation.